beancount-render/src/lib.rs

```diff
@@ -0,0 +1,413 @@
+use beancount_core::*;
+use metadata::MetaValue;
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::{fmt, fmt::Write};
+use thiserror::Error;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Default, Debug)]
+pub struct BasicRenderer {}
+
+impl BasicRenderer {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+pub fn render<W: Write>(w: &mut W, ledger: &Ledger<'_>) -> Result<(), BasicRendererError> {
+    BasicRenderer::default().render(ledger, w)
+}
+
+#[derive(Error, Debug)]
+pub enum BasicRendererError {
+    #[error("an io error occurred")]
+    Fmt(#[from] fmt::Error),
```

Comment:
The error message here seems a little inconsistent with variant name (`Fmt`).  Maybe this message could be something like "error formatting message into stream" to more closely match the origin error.  Thoughts?

https://doc.rust-lang.org/std/fmt/struct.Error.html

Reply:
Ah, that's an artifact of the `Write` change!